pkg/lib/bundle/generate.go

```diff
@@ -123,6 +125,30 @@ func GenerateFunc(directory, outputDir, packageName, channels, channelDefault st
 		return err
 	}
 
+	synthesizeErr := synthesize.Generate(directory, MetadataDir, outputDir, overwrite)
+	if synthesizeErr != nil && synthesizeErr != synthesize.CsvAlreadyExistsInBundleError && synthesizeErr != synthesize.NoCsvInferanceFileFoundError {
```

Comment:
Do we need to know the specifics about the error at the top-level? What do you think about the `Generate` function returning an error only when the failure indicates a terminal state?

```go
if err := synthesize.Generate(...); err != nil {
    return err
}
```